src/scraper/__init__.py

```diff
@@ -97,32 +215,62 @@ def get_class_and_method(self, county):
             module = importlib.import_module(module_name)
             
             # Retrieve the class from the module
-            cls = getattr(module, class_name)
+            cls = getattr(module, class_name, None)
             if cls is None:
-                print(f"Class '{class_name}' not found in module '{module_name}'.")
-                return None, None
-            
+                raise AttributeError(f"Class '{class_name}' not found in module '{module_name}'")
+
             # Instantiate the class
             instance = cls()
             
             # Retrieve the method with the specified name
             method = getattr(instance, method_name, None)
             if method is None:
-                print(f"Method '{method_name}' not found in class '{class_name}'.")
-                return instance, None
-            
+                raise AttributeError(f"Method '{method_name}' not found in class '{class_name}'")
+
             return instance, method
-        except ModuleNotFoundError:
-            print(f"Module '{module_name}' not found.")
-            return None, None
 
-    def scrape_main_page(self, base_url, odyssey_version, session, notes, logger, ms_wait):
-        # if odyssey_version < 2017, scrape main page first to get necessary data
-        if odyssey_version < 2017:
+        except (FileNotFoundError, ImportError, AttributeError) as e:
+            logger.error(f"Error dynamically loading module or retrieving class/method: {e}")
+            raise
+        except Exception as e:
+            logger.error(f"Unexpected error: {e}")
```

Comment:
No need to do this for an unexpected Exception, if the caller doesn't catch the exception it will print anyway, and if they do catch it then they probably don't want it in the logs